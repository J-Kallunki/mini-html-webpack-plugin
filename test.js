const MiniHtmlWebpackPlugin = require('./index');
const compiler = require('@webpack-contrib/test-utils');

const getConfig = (options, config = {}) =>
	Object.assign(
		{
			entry: ['./index.js'],
			plugins: [new MiniHtmlWebpackPlugin(options)],
		},
		config
	);

test('default options', async () => {
	const result = await compiler({}, getConfig());
	expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
});

test('custom title', async () => {
	const result = await compiler({}, getConfig({ context: { title: 'Pizza' } }));
	expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
});

test('custom template', async () => {
	const result = await compiler(
		{},
		getConfig({
			context: { title: 'Pizza' },
			template: ({ title }) => `<div>${title}</div>`,
		})
	);
	expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
});

test('custom filename', async () => {
	const filename = 'pizza.html';
	const result = await compiler({}, getConfig({ filename }));
	expect(result.compilation.assets[filename]._value).toMatchSnapshot();
});

// TODO: https://github.com/webpack-contrib/test-utils/issues/2
/* test('custom publicPath', async () => {
	const publicPath = '/pizza/';
	const result = await compiler({}, getConfig({}, { output: { publicPath } }));
	expect(result.compilation.assets['index.html']._value).toMatchSnapshot();
}); */
