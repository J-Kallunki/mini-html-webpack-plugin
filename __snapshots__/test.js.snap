// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom filename 1`] = `
"<!DOCTYPE html>
  <html>
    <head>
      <meta charset=\\"UTF-8\\">
      <title></title>

      
    </head>
    <body>
      <script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script>
    </body>
  </html>"
`;

exports[`custom template 1`] = `"<div>Pizza</div>"`;

exports[`custom title 1`] = `
"<!DOCTYPE html>
  <html>
    <head>
      <meta charset=\\"UTF-8\\">
      <title>Pizza</title>

      
    </head>
    <body>
      <script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script>
    </body>
  </html>"
`;

exports[`default options 1`] = `
"<!DOCTYPE html>
  <html>
    <head>
      <meta charset=\\"UTF-8\\">
      <title></title>

      
    </head>
    <body>
      <script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script>
    </body>
  </html>"
`;
